### PR TITLE
MOSIP-25716: Updated userclaims mapping

### PR DESCRIPTION
**Pull request by balaji-alluru**

### CODE REVIEW

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java [hunk 1]
```diff
@@ -32,6 +34,9 @@ public class AuthenticationContextRefUtil {
 	private static final String AMR_KEY = "amr";
 	private static final String ACR_AMR = "acr_amr";
 	private static final String IDPCLAIMSMAPPING = "idp-claims-mapping";
```
Comment (ase-101) [inline partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java:36]: is this required now?

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java [hunk 2]
```diff
@@ -86,8 +98,11 @@ private Map<List<String>, String> getAllClaims() {
 					new TypeReference<Map<String,Claims >>() {
 					});
 			Map<List<String>, String> claimsMap = new HashMap<>();
+			List<String> allowedClaims = getSupportedClaims();
 			for(Map.Entry<String,Claims> mapElement : map.entrySet()) {
-				claimsMap.put(convertStringToList(mapElement.getValue().getAttributeName()),mapElement.getKey());
+				if(allowedClaims.contains(mapElement.getKey())) {
+					claimsMap.put(convertStringToList(mapElement.getValue().getAttributeName()),mapElement.getKey());
+				}
```
Comment (ase-101) [inline partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java:105]: I see this method is returning "null" instead can we throw an exception?
With this, we don't have to worry about null checks in all the callers.

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java [hunk 3]
```diff
@@ -86,8 +98,11 @@ private Map<List<String>, String> getAllClaims() {
 					new TypeReference<Map<String,Claims >>() {
 					});
 			Map<List<String>, String> claimsMap = new HashMap<>();
+			List<String> allowedClaims = getSupportedClaims();
 			for(Map.Entry<String,Claims> mapElement : map.entrySet()) {
```
Comment (ase-101) [inline partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java:102]: if we loop on the supported list of claims, instead of all the entries available in identity-mapping.json you avoid an "if condition" in the loop.

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java (hunk 1 above)
Comment (balaji-alluru) [reply partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java]: Removed unused attributes

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java (hunk 2 above)
Comment (balaji-alluru) [reply partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java]: Done the same as suggested

File: partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java (hunk 3 above)
Comment (balaji-alluru) [reply partner/pms-common/src/main/java/io/mosip/pms/common/util/AuthenticationContextRefUtil.java]: Done the same as suggested.

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 4]
```diff
@@ -46,6 +57,10 @@ INSERT INTO pms.auth_policy
...
 
+INSERT INTO pms.auth_policy
+(id, policy_group_id, "name", descr, policy_file_id, policy_type, "version", policy_schema, valid_from_date, valid_to_date, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet-policy', 'esignet-policygroup', 'esignet-policy', 'string', '{"trustBindedAuthVerificationToken":true,"allowAuthRequestDelegation":true,"allowKycRequestDelegation":true}', 'MISP', 'string', 'https://schemas.mosip.io/v1/auth-policy', '2023-12-16 12:30:14.183', '2028-04-28 09:37:00.000', true, 'admin', '2022-12-16 12:30:14.183', 'service-account-mosip-creser-client', '2021-02-09 06:50:22.065', false, NULL) ON CONFLICT (id) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:62]: change to mpolicy-default-esignet

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 5]
```diff
@@ -55,6 +70,10 @@ INSERT INTO pms.auth_policy_h
...
 
+INSERT INTO pms.auth_policy_h
+(id, eff_dtimes, policy_group_id, name, descr, policy_file_id, policy_type, "version", policy_schema, valid_from_date, valid_to_date, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet-policy', '2020-11-13 05:58:00.000', 'esignet-policygroup', 'esignet-policy', 'esignet-policy', '{"trustBindedAuthVerificationToken":true,"allowAuthRequestDelegation":true,"allowKycRequestDelegation":true}', 'MISP', '1', 'https://schemas.mosip.io/v1/auth-policy', '2023-12-16 12:30:14.343', '2028-05-01 09:37:00.000', true, 'admin', '2022-12-16 12:30:14.343', 'service-account-mosip-creser-client', '2020-12-16 12:30:14.343', NULL, NULL) ON CONFLICT (id, eff_dtimes) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:75]: change to mpolicy-default-esignet

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 6]
```diff
@@ -29,14 +29,25 @@ INSERT INTO pms.partner
...
+VALUES('esignet-partner', '2022-12-16 12:30:14.306', 'esignet-policygroup', 'esignet-partner', 'esignet-partner', '7363038999', 'esignetpartner@mosip.io', NULL, 'mpartner-default-resident', 'MISP_Partner', 'Activated', true, 'superadmin', '2020-12-16 12:30:14.306', 'superadmin', '2020-12-16 12:30:14.306', NULL, NULL) ON CONFLICT (id, eff_dtimes) DO NOTHING;
+
 INSERT INTO pms.policy_group
 (id, "name", descr, user_id, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
 VALUES('mpolicygroup-deafult-digitalcard', 'mpolicygroup-deafult-digitalcard', 'mpolicygroup-deafult-digitalcard', 'superadmin', true, 'superadmin', '2020-12-16 12:30:14.100', 'superadmin', '2020-12-16 12:30:14.100', NULL, NULL) ON CONFLICT (id) DO NOTHING;
 
+INSERT INTO pms.policy_group
+(id, "name", descr, user_id, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet-policygroup', 'esignet-policygroup', 'esignet-policygroup', 'superadmin', true, 'superadmin', '2020-12-16 12:30:14.100', 'superadmin', '2020-12-16 12:30:14.100', NULL, NULL) ON CONFLICT (id) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:50]: change to mpolicygroup-default-esignet

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 7]
```diff
@@ -29,14 +29,25 @@ INSERT INTO pms.partner
 (id, policy_group_id, "name", address, contact_no, email_id, certificate_alias, user_id, partner_type_code, approval_status, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
 VALUES('mpartner-default-digitalcard', 'mpolicygroup-default-digitalcard', 'IITB', 'mpartner-default-digitalcard', '9232121212', 'digitalcard@mosip.io', '94d4ae61-31f0-42ca-97ae-8f4953f41fb6', 'mpartner-default-digitalcard', 'Credential_Partner', 'approved', true, 'superadmin', '2020-12-16 12:30:13.973', '110006', '2022-06-01 08:01:35.025', false, NULL) ON CONFLICT (id) DO NOTHING;
 
+INSERT INTO pms.partner
+(id, policy_group_id, "name", address, contact_no, email_id, certificate_alias, user_id, partner_type_code, approval_status, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet-partner', 'esignet-policygroup', 'IITB', 'esignet-partner', '7363038999', 'esignetpartner@mosip.io', '94d4be61-31f0-42db-97be-8f4953f41fc6', 'esignet-partner', 'MISP_Partner', 'approved', true, 'superadmin', '2020-12-16 12:30:13.973', 'superadmin', '2022-06-01 08:01:35.025', false, NULL) ON CONFLICT (id) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:34]: change to mpartner-default-esignet

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 8]
```diff
@@ -29,14 +29,25 @@ INSERT INTO pms.partner
...
 VALUES('mpartner-default-digitalcard', '2020-12-16 12:30:14.306', 'mpolicygroup-deafult-digitalcard', 'mpartner-default-digitalcard', 'mpartner-default-digitalcard', '9232121212', 'digitalcard@mosip.io', NULL, 'mpartner-default-resident', 'Credential_Partner', 'Activated', true, 'superadmin', '2020-12-16 12:30:14.306', 'superadmin', '2020-12-16 12:30:14.306', NULL, NULL) ON CONFLICT (id, eff_dtimes) DO NOTHING;
 
+INSERT INTO pms.partner_h
+(id, eff_dtimes, policy_group_id, "name", address, contact_no, email_id, certificate_alias, user_id, partner_type_code, approval_status, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet-partner', '2022-12-16 12:30:14.306', 'esignet-policygroup', 'esignet-partner', 'esignet-partner', '7363038999', 'esignetpartner@mosip.io', NULL, 'mpartner-default-resident', 'MISP_Partner', 'Activated', true, 'superadmin', '2020-12-16 12:30:14.306', 'superadmin', '2020-12-16 12:30:14.306', NULL, NULL) ON CONFLICT (id, eff_dtimes) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:42]: change to mpartner-default-esignet

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 9]
```diff
@@ -63,6 +82,9 @@ INSERT INTO pms.partner_policy
 (policy_api_key, part_id, policy_id, valid_from_datetime, valid_to_datetime, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
 VALUES('mpolicy_part_PDFCard_api', 'mpartner-default-digitalcard', 'mpolicy-default-digitalcard', '2022-02-21 07:02:26.223', '2025-12-01 05:31:00.000', true, 'admin', '2022-02-21 07:02:26.223', 'admin', '2022-02-21 07:02:26.223', false, NULL) ON CONFLICT (policy_api_key) DO NOTHING;
 
+INSERT INTO pms.partner_policy
+(policy_api_key, part_id, policy_id, valid_from_datetime, valid_to_datetime, is_active, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:86]: why is this added twice?

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql [hunk 10]
```diff
@@ -85,3 +107,7 @@ VALUES('mpartner_policy_PDFCard_req', 'mpartner-default-digitalcard', 'mpolicy-d
 INSERT INTO pms.partner_policy_request
 (id, part_id, policy_id, request_datetimes, request_detail, status_code, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
 VALUES('mpartner_policy_digitalcard_req', 'mpartner-default-digitalcard', 'mpolicy-default-digitalcard', '2022-02-21 07:02:26.292', 'mpolicy-default-digitalcard', 'approved', 'admin', '2022-02-21 07:02:26.292', 'admin', '2022-02-21 07:02:26.292', NULL, NULL) ON CONFLICT (id) DO NOTHING;
+
+INSERT INTO pms.partner_policy_request
+(id, part_id, policy_id, request_datetimes, request_detail, status_code, cr_by, cr_dtimes, upd_by, upd_dtimes, is_deleted, del_dtimes)
+VALUES('esignet_policy_req', 'esignet-partner', 'esignet-policy', '2022-02-21 07:02:26.292', 'mapping Partner to policyName', 'approved', 'superadmin', '2022-02-21 07:02:26.292', 'superadmin', '2022-02-21 07:02:26.292', NULL, NULL) ON CONFLICT (id) DO NOTHING;
```
Comment (ase-101) [inline db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:113]: change to mpartner_policy_esignet_req

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 4 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: Updated as suggested

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 5 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: updated as suggested

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 6 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: updated as suggested

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 7 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: updated as suggested

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 8 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: updated as suggested

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 9 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql:86]: That is not added twice, I forgot to updated the policy_api_key value, updated the same now

File: db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql (hunk 10 above)
Comment (balaji-alluru) [reply db_release_scripts/mosip_pms/sql/1.2.0.1_to_1.2.0.1-B2_upgrade.sql]: updated as suggested

**Review by ase-101 — APPROVED** [review]
